misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -148,6 +148,108 @@ private STNode parseArrayOfTables() {
                 identifierToken, firstCloseBracket, secondCloesBracket, STNodeFactory.createNodeList(fields));
     }
 
+    /**
+     * Parse inline table.
+     * <p>
+     * Key is an identifier. Value should be surrounded by braces. It could contain any toml top level node inside.
+     * <br>
+     * e.g. table = { key = "value" , key2 = "value2"}
+     *
+     * @return TableNode
+     */
+    private STNode parseInlineTable() {
+        STNode openBrace = parseOpenBrace();
+        STNode keyValuePairs = parseInlineKeyValuePairs();
+        STNode endBrace = parseCloseBrace();
+        return STNodeFactory.createInlineTableNode(openBrace, keyValuePairs, endBrace);
+    }
+
+    private STNode parseInlineKeyValuePairs() {
+        startContext(ParserRuleContext.INLINE_TABLE_LIST);
+        STToken nextToken = peek();
+
+        if (isEndOfInlineTable(nextToken)) {
+            STNode values = STNodeFactory.createEmptyNodeList();
+            endContext();
+            return values;
+        }
+        ArrayList<STNode> valuesList = new ArrayList<>();
+        
+        STNode value = parseInlineKeyValuePair();
```

Comment:
value is not meaningful here. maybe,
```suggestion
        STNode keyValuePair = parseInlineKeyValuePair();
```